File: scripts/file_prettier.js
const fs = require("fs");
const prettier = require("prettier");
const path = require("path");

const inputDir = path.join(__dirname, "..", "v2", "html_pages");
const outputDir = path.join(__dirname, "..", "v2", "html_pages_formatted");

async function processHtmlFiles() {
  try {
    const files = await fs.promises.readdir(inputDir);

    for (const file of files) {
      if (file.endsWith(".html")) {
        const inputFilePath = path.join(inputDir, file);
        const outputFilePath = path.join(outputDir, file);

        // Ensure `readFile` resolves completely before continuing
        const htmlContent = await fs.promises.readFile(inputFilePath, "utf8");

        const prettyHtml = prettier.format(htmlContent, { parser: "html" });
        await fs.promises.writeFile(outputFilePath, prettyHtml);
        console.log(
          `Successfully formatted ${inputFilePath} and saved to ${outputFilePath}`
        );
      }
    }

    console.log(
      `All HTML files in ${inputDir} have been formatted and saved to ${outputDir}`
    );
  } catch (error) {
    console.error(`Error processing HTML files: ${error}`);
  }
}

processHtmlFiles();

File: python/file_prettier.py
import os
from bs4 import BeautifulSoup

input_dir = os.path.join(os.path.dirname(__file__), '..', "v2", "html_pages")
output_dir = os.path.join(os.path.dirname(__file__), '..', "v2", "html_pages_formatted")

def process_html_files():
    for filename in os.listdir(input_dir):
        if filename.endswith(".html"):
            input_filepath = os.path.join(input_dir, filename)
            output_filepath = os.path.join(output_dir, filename)

            with open(input_filepath, "r") as input_file:
                html_content = input_file.read()

            soup = BeautifulSoup(html_content, "html.parser")
            pretty_html = soup.prettify()

            with open(output_filepath, "w") as output_file:
                output_file.write(pretty_html)

            print(f"Formatted {input_filepath} and saved to {output_filepath}")

    print("All HTML files have been formatted and saved.")

process_html_files()